core/hw/maple/maple_cfg.cpp

```diff
@@ -68,14 +64,19 @@ void mcfg_Create(MapleDeviceType type,u32 bus,u32 port)
 
 void mcfg_CreateDevices()
 {
-#if DC_PLATFORM == DC_PLATFORM_DREAMCAST
-	mcfg_Create(MDT_SegaController,0,5);
-
-	mcfg_Create(MDT_SegaVMU,0,0);
-	mcfg_Create(MDT_SegaVMU,0,1);
-#else
-	mcfg_Create(MDT_NaomiJamma, 0, 5);
-#endif
+	for(int port = 0; port < MAPLE_NUM_PORTS; port++)
```

Comment:
Call `cfgLoadStr`/Int` here, you mean? I'd rather not do that and rather have a settings object that's decoupled to some degree from the actual config file loading/parsing.

That'd make clean libretro port almost impossible (since that won't use the usual config file and use some keys from `~/.config/retroarch/retroarch-core-options.cfg` instead).  Also, the config file *can* be platform-dependent IMO (like using [`SharedPreferences`](http://developer.android.com/reference/android/content/SharedPreferences.html) on Android for example).

What we could do: Make all the `cfgLoadFoo`/`cfgSaveFoo` a more abstract interface that supports one of these "backends" depending on platform.

Alternative: We could drop the `enabled` flag and introduce an `enable()` method instead, that calls `mcfg_create()` (don't know if that's possible at runtime though).